lib/task/migrate/arUpgradeSqlTask.class.php

```diff
@@ -149,88 +150,117 @@ protected function execute($arguments = array(), $options = array())
     // Find all the upgrade classes in lib/task/migrate
     $version = $this->initialVersion;
 
-    foreach (sfFinder::type('file')
-              ->maxdepth(0)
-              ->sort_by_name()
-              ->name('arUpgrader*.class.php')
-              ->in(sfConfig::get('sf_lib_dir').'/task/migrate') as $filename)
+    if (!empty($options['number']))
     {
-      $className = preg_replace('/.*(arUpgrader\d+).*/', '$1', $filename);
-      $class = new $className;
-
-      if ($class::INIT_VERSION > $version)
-      {
-        continue;
-      }
-
-      try
+      foreach (explode(',', $options['number']) as $number)
```

Comment:
I think it would be better to factor the `$number` option code into a separate method to aid readability.